pinot-core/src/main/java/org/apache/pinot/core/common/MinionConstants.java

```diff
@@ -67,16 +68,22 @@ private MinionConstants() {
     public static final String GRANULARITY_KEY = "granularity";
 
     // Rollup aggregate function related configs
-    public static final String AGGREGATE_KEY_PREFIX = "aggregate";
+    public static final String AGGREGATION_TYPE_KEY_SUFFIX = ".aggregationType";
 
     // Merge properties related configs
-    public static final String MERGE_KEY_PREFIX = "merge";
-    public static final String BUFFER_TIME = "bufferTime";
+    public static final String BUFFER_TIME_PERIOD = "bufferTimePeriod";
+    public static final String BUCKET_TIME_PERIOD = "bucketTimePeriod";
+    public static final String ROUND_BUCKET_TIME_PERIOD = "roundBucketTimePeriod";
     public static final String MAX_NUM_RECORDS_PER_SEGMENT = "maxNumRecordsPerSegment";
     public static final String MAX_NUM_RECORDS_PER_TASK = "maxNumRecordsPerTask";
 
     // Segment name generator related configs
-    public static final String MERGED_SEGMENT_NAME_KEY = "mergedSegmentNameKey";
+    public static final String SEGMENT_NAME_PREFIX_KEY = "segmentNamePrefix";
+    public static final String SEGMENT_NAME_PREFIX = "merged";
+
+    // Window start and window end set by task generator
+    public static final String WINDOW_START_MS_KEY = "windowStartMs";
```

Comment:
For merge/rollup task, we should not use the window start/end, but use the partition bucket instead. Different from the realtime-to-offline task, we need to keep all the records (cannot filter the records) for merge/rollup task.

Reply:
We will not filter the records, while the start/end time are useful to mark the segment as merged by setting customMap as `{mergeRollupTask.bucketGranularity : granularityLevel}`. Say we have segments: `s1_0719_0720`, `s2_0720_0720` and we use 1 day bucket time. The first run merge `s1_0719_0720` to `merged_s3_0719_0719` and `merged_s4_0720_0720`. `merged_s3_0719_0719` will be marked since it's in the target window and won't be picked for the next merge task. `merged_s4_0720_0720` will not be marked. The second merge task will pick `merged_s4_0720_0720` and `s2_0720_0720`.

Reply:
I think that it makes sense that we need to know about the current window's start & end time in order to mark the segments. 

What happens if some segments that are older than the watermark get pushed? (e.g. watermark=`0720`, segment=`0715_0715`) I think that we haven't covered this case on how we handle it. @Jackie-Jiang @jtao15

Reply:
We assume all segments before the watermark are merged, so it won't have a chance to be merged. Hmm, why would the user want to upload some old data? If they want to backfill 07/15 data, they should replace all segments on 07/15. For now, we are not covering the case if they backfill a large number of segments on 07/15.

Reply:
There's a valid use case:

1. user pushes data for 07/15
2. pinot merges segments for 07/15
3. user finds out the issue and backfill 07/15 later

4. it's good if pinot merges backfilled segments for 07/15 but we won't do it with the current design since those segments passed the watermark. As we discussed today, one solution is to open up the way for triggering ad-hoc merge-rollup job for a specific window.